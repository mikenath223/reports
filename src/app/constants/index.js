export const pieColors = ["#A259FF", "#F24E1E", "#FFC107", "#6497B1"];
