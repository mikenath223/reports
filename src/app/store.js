import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import projectsReducer from "./slices/projectsSlice";
import formReducer from "./slices/formSlice";
import userReducer from "./slices/userSlice";
import gatewayReducer from "./slices/gatewaySlice";
import reportsReducer from "./slices/reportsSlice";

const middlewares = [];
if (process.env.NODE_ENV === `development`) {
  middlewares.push(logger);
}

export const store = configureStore({
  reducer: {
    projects: projectsReducer,
    form: formReducer,
    users: userReducer,
    gateways: gatewayReducer,
    reports: reportsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(middlewares),
});
