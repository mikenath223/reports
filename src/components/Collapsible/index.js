import { Accordion, AccordionDetails, AccordionSummary } from "@mui/material";
import PropTypes from "prop-types";

export default function AccordionElem({
  header,
  expanded,
  setExpanded,
  children,
}) {
  return (
    <Accordion className="accordion" expanded={expanded} onChange={setExpanded}>
      <AccordionSummary
        expandIcon={null}
        aria-controls="panel1a-content"
        id="panel1a-header"
      >
        {header}
      </AccordionSummary>
      <AccordionDetails>{children}</AccordionDetails>
    </Accordion>
  );
}

AccordionElem.propTypes = {
  header: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  expanded: PropTypes.bool,
  setExpanded: PropTypes.func,
};
