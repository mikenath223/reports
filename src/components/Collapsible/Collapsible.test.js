import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "app/store";
import Collapsible from ".";

describe("Collapsible", function () {
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <Collapsible />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
