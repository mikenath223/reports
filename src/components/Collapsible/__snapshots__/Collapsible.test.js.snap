// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters accordion css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-controls="panel1a-content"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
      id="panel1a-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      />
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            class="MuiAccordion-region"
            id="panel1a-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
